### DISCUSSION TITLE
Webchat - New Message Badge

### DISCUSSION THREAD

**Discussion by Fazle-Jahangir** [twilio-labs/paste#2254] [General] [answered]
Hi, 

I would appreciate your input and guidance on the proposed design for adding a "new message" badge for webchat. 

When the webchat icon is closed a badge will display the number of new unread messages received. However, this notification is placed on the left-hand side of the icon as the icon itself is anchored to the bottom-right. 

![New Message Badge](https://user-images.githubusercontent.com/100706574/156219790-9487b94f-5bbe-4645-888e-665c8f4758f9.jpg)

**Comment by shleewhite** [comment, marked as answer]
Hi @Fazle-Jahangir! 

For the badge, we are currently implementing [version 2 of badge](https://docs.google.com/document/d/1O-8DVZQMMVeCCNjmsFpg2g7HvszmINsCRemdKSDVmCQ/edit#heading=h.wrc83btl6de1) that includes guidance for counter badges. The colors you have now are not a supported badge variant, so we would recommend using the `neutral` or `decorative10` variants.
![Screen Shot 2022-03-02 at 1 08 30 PM](https://user-images.githubusercontent.com/8482334/156435198-33ab5b4c-7f17-4501-b63a-877f3df61eea.png)

We are also currently doing visual design work for chat UIs across twilio, you can see [our current WIP](https://www.figma.com/file/cSdKlqoTbxoAUmnrYkfFWl/UI-Webchat-Kit?node-id=0%3A1)
![Screen Shot 2022-03-02 at 1 10 45 PM](https://user-images.githubusercontent.com/8482334/156436150-b18c6301-743b-4450-806e-067a8c266695.png)

Please let me know if you have any other questions.
